File: src/main/java/org/fountainmc/api/world/block/Chest.java
package org.fountainmc.api.world.block;

import org.fountainmc.api.Direction;

public interface Chest extends BlockState, DirectionalBlock {

    Direction getDirection();

    /**
     * Get a clone of this chest with the given direction
     *
     * @param direction the new direction
     * @return the new chest
     */
    Chest withDirection(Direction direction);

    @Override
    default boolean mayBeVertical() {
        return false;
    }

}

File: src/main/kotlin/org/fountainmc/api/world/block/Chest.kt
package org.fountainmc.api.world.block

import org.fountainmc.api.Direction

interface Chest : BlockState, DirectionalBlock {

    val direction: Direction

    /**
     * Get a clone of this chest with the given direction

     * @param direction the new direction
     * *
     * @return the new chest
     */
    override fun withDirection(direction: Direction): Chest

    override fun mayBeVertical(): Boolean {
        return false
    }

}
